langchain/src/chains/combine_docs_chain.ts

```diff
@@ -215,3 +218,152 @@ export class MapReduceDocumentsChain
     };
   }
 }
+
+export interface RefineDocumentsChainInput extends StuffDocumentsChainInput {
+  refineLLMChain: LLMChain;
+  documentPrompt: BasePromptTemplate;
+}
+
+/**
+ * Combine documents by doing a first pass and then refining on more documents.
+ * @augments BaseChain
+ * @augments RefineDocumentsChainInput
+ */
+export class RefineDocumentsChain
+  extends BaseChain
+  implements RefineDocumentsChainInput
+{
+  llmChain: LLMChain;
+
+  inputKey = "input_documents";
+
+  outputKey = "output_text";
+
+  documentVariableName = "context";
+
+  initialResponseName = "existing_answer";
+
+  refineLLMChain: LLMChain;
+
+  get defaultDocumentPrompt(): BasePromptTemplate {
+    return new PromptTemplate({
+      inputVariables: ["page_content"],
+      template: "{page_content}",
+    });
+  }
+
+  documentPrompt = this.defaultDocumentPrompt;
+
+  get inputKeys() {
+    return [this.inputKey, ...this.refineLLMChain.inputKeys];
+  }
+
+  constructor(fields: {
+    llmChain: LLMChain;
+    refineLLMChain: LLMChain;
+    inputKey?: string;
+    outputKey?: string;
+    documentVariableName?: string;
+    documentPrompt?: BasePromptTemplate;
+    initialResponseName?: string;
+  }) {
+    super();
+    this.llmChain = fields.llmChain;
+    this.refineLLMChain = fields.refineLLMChain;
+    this.documentVariableName =
+      fields.documentVariableName ?? this.documentVariableName;
+    this.inputKey = fields.inputKey ?? this.inputKey;
+    this.documentPrompt = fields.documentPrompt ?? this.documentPrompt;
+    this.initialResponseName =
+      fields.initialResponseName ?? this.initialResponseName;
+  }
+
+  _constructInitialInputs(docs: Document[], rest: Record<string, unknown>) {
+    const baseInfo: Record<string, unknown> = {
+      page_content: docs[0].pageContent,
+      ...docs[0].metadata,
+    };
+    const documentInfo: Record<string, unknown> = {};
+    this.documentPrompt.inputVariables.forEach((value) => {
+      documentInfo[value] = baseInfo[value];
+    });
+
+    const baseInputs: Record<string, unknown> = {
+      [this.documentVariableName]: this.documentPrompt.format({
+        ...documentInfo,
+      }),
+    };
+    const inputs = { ...baseInputs, ...rest };
+    return inputs;
+  }
+
+  _constructRefineInputs(doc: Document, res: ChainValues) {
+    const baseInfo: Record<string, unknown> = {
+      page_content: doc.pageContent,
+      ...doc.metadata,
+    };
+    const documentInfo: Record<string, unknown> = {};
+    this.documentPrompt.inputVariables.forEach((value) => {
+      documentInfo[value] = baseInfo[value];
+    });
+    const baseInputs: Record<string, unknown> = {
+      [this.documentVariableName]: this.documentPrompt.format({
+        ...documentInfo,
+      }),
+    };
+    const inputs = { [this.initialResponseName]: res, ...baseInputs };
```

Comment:
res right now is ChainValues, but it should be a just a strings

since this is an llm chain, there should only be one value in there. i think we actually expose `predict` in the JS LLMChain, which returns a string (so you could try using that)

Reply:
**Resolution:** 
Used `predict` instead of call. 
The `_call` function now returns the string response `res` using the `outputKey`.